tests/rptest/tests/resource_limits_test.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.clients.rpk import RpkTool, RpkException
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import ResourceSettings
+from rptest.services.cluster import cluster
+
+
+class ResourceLimitsTest(RedpandaTest):
+    """
+    Tests the limits that are imposed on partition counts during topic
+    creation, to prevent users from overwhelming available system
+    resources with too many partitions.
+    """
+    def setUp(self):
+        # Override parent setUp so that we don't start redpanda until each test,
+        # enabling each test case to customize its ResourceSettings
+        pass
+
+    @cluster(num_nodes=3)
```

Comment:
Can you please add a test case for setting these config params to zero, or show me where we prevent that?